modules/fedimint-wallet-client/src/deposit.rs

```diff
@@ -0,0 +1,368 @@
+                    move |dbtx, txout_proof, old_state| {
+                        Box::pin(transition_btc_tx_confirmed(
+                            dbtx,
+                            global_context.clone(),
+                            old_state,
+                            txout_proof,
+                        ))
+                    },
+                )]
+            }
+            DepositStates::Claiming(_) => {
+                vec![]
+            }
+            DepositStates::TimedOut(_) => {
+                vec![]
+            }
+        }
+    }
+
+    fn operation_id(&self) -> OperationId {
+        self.operation_id
+    }
+}
+
+async fn await_crated_btc_transaction_submitted(
+    context: WalletClientContext,
+    tweak: KeyPair,
+) -> (bitcoin::Transaction, u32) {
+    let script = context
+        .wallet_descriptor
+        .tweak(&tweak.public_key().to_x_only_pubkey(), &context.secp)
+        .script_pubkey();
+    loop {
+        match get_script_history(&context.esplora_server, &script).await {
+            Ok(received) => {
+                // TODO: fix
+                if received.len() > 1 {
+                    warn!("More than one transaction was sent to deposit address, only considering the first one");
+                }
+
+                if let Some(transaction) = received.into_iter().next() {
+                    let out_idx = transaction
+                        .output
+                        .iter()
+                        .enumerate()
+                        .find_map(|(idx, output)| {
+                            if output.script_pubkey == script {
+                                Some(idx as u32)
+                            } else {
+                                None
+                            }
+                        })
+                        .expect("TODO: handle invalid tx returned by API");
+
+                    return (transaction, out_idx);
+                } else {
+                    trace!("No transactions received yet for script {script:?}");
+                }
+            }
+            Err(e) => {
+                warn!("Error fetching transaction history for {script:?}: {e}");
+            }
+        }
+
+        sleep(Duration::from_secs(10)).await;
+    }
+}
+
+async fn get_script_history(
+    esplora_server: &str,
+    script: &Script,
+) -> anyhow::Result<Vec<bitcoin::Transaction>> {
+    let esplora_client = esplora_client(esplora_server)?;
+    let script = script.clone();
+
+    let transactions = esplora_client
+        .scripthash_txs(&script, None)
+        .await?
+        .into_iter()
+        .map(|tx| tx.to_tx())
+        .collect::<Vec<_>>();
+
+    Ok(transactions)
+}
+
+async fn transition_tx_seen(
+    old_state: DepositStateMachine,
+    btc_transaction: bitcoin::Transaction,
+    out_idx: u32,
+) -> DepositStateMachine {
+    let DepositStateMachine {
+        operation_id,
+        state: old_state,
+    } = old_state;
+
+    match old_state {
```

Comment:
Nit: I thought the linter would yell at you if a match statement is used with only one option (instead of using `if let`

Reply:
I actually use the unexpected state for logging, so it can't be a if-let